src/components/ProjectSearch/TextSearch.js

```diff
@@ -11,14 +11,45 @@ import "./TextSearch.css";
 
 import { getRelativePath } from "../../utils/sources-tree";
 
+// This finds a node that matches (has the same values
+// for specific properties) as the node passed in.
+
+function getMatchingFocusNode(results, node) {
+  if (node === null) {
+    return null;
+  }
+  let matchingNode = null;
+  results.some(result => {
```

Comment:
do we expect only one matching node in the collection? perhaps we can use `find` here instead and do
```
const matchingNode = results.find(....) // same code as before, but return `result` rather than assigning matchingNode
```

Reply:
it looks like `.find` would not work here because there is an `else` part searches the `matches` inside the `result` rather than returning the `result`

Reply:
@bomsy I think @codehag is right here. You can use `find` and it would look like so

```js
// Or just return this call.
const matchingNode = results.find(result => {
  if (node.filepath) {
    if (node.sourceId === result.sourceId) {
        return result;
    }
  } else {
    return result.matches.find(match => {
      if (
        node.sourceId === match.sourceId &&
        node.column === match.column &&
        node.line === match.line
      ) {
        return match;
      }
    });
  }
})

Reply:
yeah .. i tried it and it doesn't seem to work

Reply:
```js
return result.matches.find(match => { ...
```
this line will return undefined